examples/datasource-streaming-backend-websocket/streaming-backend-websocket-plugin/pkg/plugin/datasource.go

```diff
@@ -172,6 +167,32 @@ func (d *Datasource) PublishStream(context.Context, *backend.PublishStreamReques
 }
 
 func (d *Datasource) RunStream(ctx context.Context, req *backend.RunStreamRequest, sender *backend.StreamSender) error {
+	chunks := strings.Split(req.Path, "/")
+	if len(chunks) < 3 {
+		return fmt.Errorf("invalid path: %s", req.Path)
+	}
+
+	q := Query{
+		LowerLimit: chunks[1],
+		UpperLimit: chunks[2],
+	}
```

Comment:
this may work but I see that there ris a req.Data that seems like a better place to put query inputs. See for example how Loki is using it:

https://github.com/grafana/grafana/blob/main/pkg/tsdb/loki/streaming.go#L68

Reply:
To be able to set `req.Data` I had to add frontend code. I've added a new implementation to `query` to create the stream. That made the `Query` implementation in the backend unnecessary to create the stream, so I removed it, which made the implementation of `backend.QueryDataHandler` unnecessary as well.